### DISCUSSION TITLE
Is the container booted when we run the container?

### DISCUSSION THREAD

**Discussion by Heng-Zhou** [sylabs/singularity#2558] [Q&A] [answered]
Here I use "run" in an informal way which represents two cases. 

First, when we run commands in `%post` section during building, has the base OS booted, i.e., the running environment of commands in `%post` are like in the command line of the base OS after it is booted normally (not in a container)?

Second, when we shell into the container, has the base OS completed booting before showing the `Singularity>` prompt? Thank you.

**Comment by dtrudg** [comment, marked as answer]
There is generally no concept of 'booting' a Singularity container. A container shares the kernel of the host system, so there is no low level boot up of drivers etc.

A container does not usually have an init system (generally systemd), nor does it run the various services an OS will when booted on a physical or virtual machine.

When you run commands in `%post`, the only environment is whatever has been setup by earlier commands, or the base image in your `From:` line.

When you `shell` into a container, only the shell process is started inside the container. Generally nothing else is 'booted'.
